### PR TITLE
A sound pool implementation for iOS

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

File: sky/services/media/ios/sound_pool_impl.mm [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+SoundPoolImpl::~SoundPoolImpl() {
+  [sound_pool_ release];
+}
+
+static base::FilePath TemporaryFilePath() {
+  char temp[256] = {0};
+  NSString* tempDirectory = NSTemporaryDirectory();
+  snprintf(temp, sizeof(temp), "%spool.XXXXXX", tempDirectory.UTF8String);
+  base::FilePath path(mktemp(temp));
```
Comment (abarth) [inline sky/services/media/ios/sound_pool_impl.mm]: When do we delete this temp file?

**Comment by abarth** [comment]
LGTM

It seems slightly awkward to steam the data to a temp file.  I suspect the data already exists in a file on the device.  We might want to change the API so that we can avoid having to copy the data from one file to another.

File: sky/services/media/ios/sound_pool_impl.mm (hunk 1 above)
Comment (chinmaygarde) [inline sky/services/media/ios/sound_pool_impl.mm]: Fixed

**Comment by abarth** [comment]
LGTM

**Pull request merged**